src/preprocess/query_generator.py

```diff
@@ -167,6 +170,160 @@ def generate_anonymize_user_queries(new_admin,old_admin, exclude_users, f, prepr
+    DELETE FROM  intepretation_likedby where interpretationid in (select interpretationid from interpretation where organisationunitid in (select * from orgUnitsToDelete));
+    DELETE FROM  interpretationuseraccesses where interpretationid in (select interpretationid from interpretation where organisationunitid in (select * from orgUnitsToDelete));
+    DELETE FROM  interpretation where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM eventvisualization_organisationunits WHERE organisationunitid      IN (SELECT * FROM orgUnitsToDelete);
+    DELETE FROM  lockexception where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  mapview_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  organisationunitattributevalues where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  program_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM programmessage_emailaddresses    WHERE programmessageemailaddressid     IN (SELECT * FROM rm_programmessage);
+    DELETE FROM programmessage_deliverychannels  WHERE programmessagedeliverychannelsid IN (SELECT * FROM rm_programmessage);
+    DELETE FROM programmessage                   WHERE id                               IN (SELECT * FROM rm_programmessage);
+    DELETE FROM  reporttable_organisationunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  userdatavieworgunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  usermembership where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  userteisearchorgunits where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM  validationresult where organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM completedatasetregistration where sourceid in (select * from orgUnitsToDelete);
+    DELETE FROM configuration WHERE selfregistrationorgunit in (select * from orgUnitsToDelete);
+    DELETE FROM minmaxdataelement WHERE sourceid in (select * from orgUnitsToDelete);
+    DELETE FROM visualization_organisationunits WHERE organisationunitid in (select * from orgUnitsToDelete);
+    DELETE FROM organisationunit WHERE organisationunitid in (select * from orgUnitsToDelete);
+    DROP MATERIALIZED VIEW if exists orgUnitsToDelete  CASCADE;
+    
+    """)
+
+
+def generate_delete_org_unit_level_rules(level, f):
+    write(f,"""
+    --remove organisationUnits -- org unit
+    DROP MATERIALIZED VIEW if exists orgUnitsToDelete CASCADE;
+    """)
+    write(f,
+        " CASCADE;"
+        "CREATE MATERIALIZED VIEW orgUnitsToDelete AS select organisationunitid from organisationunit where hierarchylevel > {} ; \n".format(
+            level))
+    create_orgunittoberemoved_indixes(f)
+    delete_org_unit_data_in_view(f)
+    delete_org_units_in_view(f)
+
+
+def create_orgunittoberemoved_indixes(f):
```

Comment:
(very minor typo) The term is "indexes".

Also, the org units to delete is called `orgUnitsToDelete` elsewhere, so maybe `create_orgUnitsToDelete_indices()` (or without the caps) would be more appropriate. In any case, I'd probably call it something like `create_views_and_indices()`, but I'll comment more on the global review!